api/src/backend/tasks/tasks.py

```diff
@@ -251,84 +255,106 @@ def generate_outputs(scan_id: str, provider_id: str, tenant_id: str):
         logger.info(f"No findings found for scan {scan_id}")
         return {"upload": False}
 
-    # Initialize the prowler provider
-    prowler_provider = initialize_prowler_provider(Provider.objects.get(id=provider_id))
+    provider_obj = Provider.objects.get(id=provider_id)
```

Comment:
Could we take this opportunity to wrap this code out of the task logic itself and move it to `export.py`? To improve readability throughout the defined tasks and also allow seamless migrations to another Task Runner framework if required in the future.

I know there are old tasks that should be migrated to that pattern as well, just asking.

Reply:
This implies a lot of changes in the tests and since we know that this module has to be refactored I think it is more appropriate to do it all in a PR. The tests added in this PR have been done because we have no date for this refactor

Reply:
Could you create a ticket for it?